src/EReg/PIDController.cpp

```diff
@@ -0,0 +1,26 @@
+#include "PIDController.h"
+
+PIDController::PIDController(double p, double i, double d, double minSpeed, double maxSpeed) {
+    p_, i_, d_ = p, i, d;
+    minOutput_, maxOutput_ = minSpeed, maxSpeed;
+    lastUpdate_ = micros();
+    previousError_ = 0;
+    intError_ = 0;
+}
+
+double PIDController::update(double error) {
+    float curr_time = micros();
+    float dt = curr_time - lastUpdate_;
+    double output = p_*error + d_*(error-previousError_)/dt;
+
+    intError_ += error * dt;
+    output += i_ * intError_;
+
+    output = output < minOutput_ ? minOutput_ : output;
```

Comment:
why not `min(max(output, minOutput), maxOutput)`?

Reply:
I saw that and didn't want to figure it out but yeah I'll put that